src/main/java/technology/dice/dicefairlink/iterators/RandomisedCyclicIterator.java

```diff
@@ -34,6 +35,17 @@ public boolean hasNext() {
     return !elements.isEmpty();
   }
 
+  public boolean hasSameContent(final Collection<T> externalElements) {
+    if (externalElements == null || externalElements.size() != elements.size()) {
+      return false;
+    }
+    return externalElements.stream().unordered().parallel().allMatch(foundInInternalCollection());
```

Comment:
Aurora supports a maximum of 15 replicas at the time of writing.
Using a `parallel` stream is probably slower than not using.

Please remove

Reply:
Removed.